examples/mllib_mlp.py

```diff
@@ -1,66 +1,134 @@
+from keras.layers.core import Dense, Dropout, Activation, Flatten
+from keras.layers.convolutional import Convolution2D, MaxPooling2D
+from keras import backend as K
 
-from elephas.spark_model import SparkMLlibModel
-from elephas.utils.rdd_utils import to_labeled_point
+from elephas.spark_model import SparkMLlibModel, SparkModel
+from elephas.utils.rdd_utils import to_labeled_point, to_simple_rdd
 
 from pyspark import SparkContext, SparkConf
 
 # Define basic parameters
-batch_size = 64
+batch_size = 128
 nb_classes = 10
-epochs = 3
+epochs = 12
+
+# input image dimensions
+img_rows, img_cols = 28, 28
+# number of convolutional filters to use
+nb_filters = 32
+# size of pooling area for max pooling
+nb_pool = 2
+# convolution kernel size
+nb_conv = 3
 
 # Load data
 (x_train, y_train), (x_test, y_test) = mnist.load_data()
 
+
+# Flatten the data, MLP doesn't use the 2D structure of the data. 784 = 28*28
 x_train = x_train.reshape(60000, 784)
 x_test = x_test.reshape(10000, 784)
+
+#see http://cs231n.github.io/convolutional-networks/#overview
+# if K.image_data_format() == 'channels_first':
+#     x_train = x_train.reshape(x_train.shape[0], 1, img_rows, img_cols)
+#     x_test = x_test.reshape(x_test.shape[0], 1, img_rows, img_cols)
+#     input_shape = (1, img_rows, img_cols)
+# else:
+#     x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
+#     x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
+#     input_shape = (img_rows, img_cols, 1)
+
+# Make the value floats in [0;1] instead of int in [0;255]
 x_train = x_train.astype("float32")
 x_test = x_test.astype("float32")
 x_train /= 255
 x_test /= 255
+# Display the shapes to check if everything's ok
+print('x_train shape:', x_train.shape)
 print(x_train.shape[0], 'train samples')
 print(x_test.shape[0], 'test samples')
 
-# Convert class vectors to binary class matrices
+# convert class vectors to binary class matrices (ie one-hot vectors)
 y_train = np_utils.to_categorical(y_train, nb_classes)
 y_test = np_utils.to_categorical(y_test, nb_classes)
 
 model = Sequential()
-model.add(Dense(128, input_dim=784))
-model.add(Activation('relu'))
+model.add(Dense(512, activation='relu', input_shape=(784,)))
 model.add(Dropout(0.2))
-model.add(Dense(128))
-model.add(Activation('relu'))
+model.add(Dense(512, activation='relu'))
 model.add(Dropout(0.2))
-model.add(Dense(10))
-model.add(Activation('softmax'))
+model.add(Dense(nb_classes, activation='softmax'))
+
+# model = Sequential()
+# #For an explanation on conv layers see http://cs231n.github.io/convolutional-networks/#conv
+# #By default the stride/subsample is 1
+# #border_mode "valid" means no zero-padding.
+# #If you want zero-padding add a ZeroPadding layer or, if stride is 1 use border_mode="same"
+# model.add(Convolution2D(nb_filters, nb_conv, nb_conv,
+#                         border_mode='valid',
+#                         input_shape=input_shape))
+# model.add(Activation('relu'))
```

Comment:
I don't fully understand why you change so much in this example, but it's fine with me. Please remove all (!) the commented code you left throughout this PR, ok? thanks

Reply:
this is just my test sample. sorry for make you confuse.